### PR TITLE
Replace usage of deprecated $canonicalize phpunit parameter

### PR DESCRIPTION
**Pull request by G-Rath**
More deprecated things.

In other news, I found the [PHPUnit issue on github](https://github.com/sebastianbergmann/phpunit/issues/3341) 😃

### CODE REVIEW

File: exercises/allergies/allergies_test.php [hunk 1]
```diff
@@ -26,7 +30,7 @@ public function testAllergiesToOneAllergen($allergicTo)
         $otherAllergen = array_filter(Allergen::allergenList(), function ($allergen) use ($allergicTo) {
             return $allergen != $allergicTo;
         });
-        $self = $this;
+        $self          = $this;
         array_map(function ($allergen) use ($allergies, $self) {
             $self->assertFalse($allergies->isAllergicTo($allergen));
```
Comment (petemcfarlane) [inline exercises/allergies/allergies_test.php]: hmm this is weird, passing `$self` through here, not sure why it isn't just `$this->assertFalse()` inside the closure 🤔

Comment (rossbearman) [reply exercises/allergies/allergies_test.php]: Prior to PHP 5.4 `$this` wasn't bound automatically inside anonymous functions, so it was possibly written with that in mind? The test was originally written in 2016, when 5.4 had only just reached it's end of life and keeping PHP up to date wasn't as common.

It should probably be removed now, however, as 5.4 has been EOL'd for four years.

Comment (G-Rath) [reply exercises/allergies/allergies_test.php]: Let me know if you guys want me to remove this :)

Also: Hi Ross! :P

Comment (rossbearman) [reply exercises/allergies/allergies_test.php]: I think it makes sense to remove it, as it's bad practice now and the tests are something students should be looking at while they're working through a problem.

Thoughts @petemcfarlane?

Comment (petemcfarlane) [reply exercises/allergies/allergies_test.php]: Yeah 💯

**Comment by DerTee** [comment]
@G-Rath Since the assertEqualsCanonicalize thing was fixed in a different PR: Could you rebase this PR on the current master and check what is left?
I think at least your are other two commits are still valid and should be merged.

File: exercises/allergies/allergies_test.php (hunk 1 above)
Comment (G-Rath) [reply exercises/allergies/allergies_test.php]: If you guys want this change, could you please request it? (i.e review with "changes requested")

/cc @DerTee

**Comment by G-Rath** [comment]
Sure thing!

**Comment by G-Rath** [comment]
@DerTee I've merged this branch with master - hopefully it's how you want it. I had a bit of trouble with rebasing & such (I'm still learning about working across forks), so let me know if theres any problems with it.

Worse case, I can remake the branch by just cherry picking the commits out.

**Comment by DerTee** [comment]
@G-Rath No, a merge just clutters the history unnecessarily in this case. But a rebase with conflicts can be a major pain, that's true :)

I think you are right, doing a cherry pick is probably the smarter move (maybe in a new PR if it's too much hassle to change this one, I don't know what's better).

File: exercises/allergies/allergies_test.php [hunk 2]
```diff
@@ -26,7 +30,7 @@ public function testAllergiesToOneAllergen($allergicTo)
         $otherAllergen = array_filter(Allergen::allergenList(), function ($allergen) use ($allergicTo) {
             return $allergen != $allergicTo;
         });
-        $self = $this;
+        $self          = $this;
```
Comment (petemcfarlane) [inline exercises/allergies/allergies_test.php]: Please can we remove this temporary variable `$self`, as it's no longer necessary

**Pull request merged**